src/main/java/core/characters/CharacterManager.java

```diff
@@ -1,80 +1,145 @@
 package core.characters;
 
+import java.lang.reflect.Array;
 import java.util.ArrayList;
-
-// Need to import InputManager and ControlState
+import java.util.HashMap;
+import java.util.Objects;
 
 public class CharacterManager {
 
     /*
      * Use case class that handles updating instances of GameCharacter based on inputs from the InputHandler
      * @param character: An instance of GameCharacter being updated
+     * @param type: Either Player or AI so the input handler can distinguish between usages
      * TODO: Change item to type Item when implemented
-     * TODO: Import InputManager and ControlState
      * */
 
-    private GameCharacter character;
+    public HashMap<Integer, GameCharacter> characterEntities;
+    private ArrayList<Integer> keylist;
+
+    public CharacterManager() {
+        this.characterEntities = new HashMap<Integer, GameCharacter>();
+        this.keylist = new ArrayList<Integer>();
+    }
 
-    public CharacterManager(GameCharacter character) {
-        this.character = character;
+    public void addCharacter(GameCharacter character) {
+        /*
+        * Player will always be hashed as 0
+        * Defenders will be hashed as even ints
+        * Enemies will be hashed as odd ints
+        * TODO: Refactor
+        * */
+        int maxElement = this.keylist.get(this.keylist.size() - 1);
+
+        if (Objects.equals(character.team, "player")) {
+            this.characterEntities.put(0, character);
+            this.keylist.add(0);
+        } else if (Objects.equals(character.team, "defender")) {
+            if (maxElement % 2 == 0) {
+                this.characterEntities.put(maxElement + 2, character);
+                this.keylist.add(maxElement + 2);
+            } else {
+                this.characterEntities.put(maxElement + 1, character);
+                this.keylist.add(maxElement + 1);
+            }
+        } else {
+            if (maxElement % 2 == 0) {
+                this.characterEntities.put(maxElement + 1, character);
+                this.keylist.add(maxElement + 1);
+            } else {
+                this.characterEntities.put(maxElement + 2, character);
+                this.keylist.add(maxElement + 2);
+            }
+        }
     }
 
-    public void moveCharacter() {
-        // Updates the characters position
+    public void moveCharacter(int id, float newX, float newY) {
+        //Updates the position of the character
+        for (var i : this.characterEntities.entrySet()) {
```

Comment:
do we have to loop thru the whole set?

should use `.get` if available or use a collection where it's easy to get by key